utils/lib/rr_graph/channel.py

```diff
@@ -243,318 +268,516 @@ def __str__(self):
-        C(C,1)
+        >>> g.create_track(Track((4, 5), (6, 5), None, "T"))
+        T(T,1)
         >>> g[(3,5)]
-        [C(A,0), None]
+        [T(A,0), None]
         >>> g[(4,5)]
-        [C(B,0), C(C,1)]
+        [T(B,0), T(T,1)]
         >>> g[(6,5)]
-        [C(B,0), C(C,1)]
+        [T(B,0), T(T,1)]
         >>> # Adding a channel which overlaps, but is a row over
-        >>> g.add_channel(Channel((4, 6), (6, 6), None, "D"))
-        C(D,0)
+        >>> g.create_track(Track((4, 6), (6, 6), None, "D"))
+        T(D,0)
         >>> g[(4,5)]
-        [C(B,0), C(C,1)]
+        [T(B,0), T(T,1)]
         >>> g[(4,6)]
-        [C(D,0)]
+        [T(D,0)]
         >>> # Adding fourth channel which overlaps both the first
         >>> # and second+third channel
-        >>> g.add_channel(Channel((2, 5), (5, 5), None, "E"))
-        C(E,2)
+        >>> g.create_track(Track((2, 5), (5, 5), None, "E"))
+        T(E,2)
         >>> g[(1,5)]
-        [C(A,0), None, None]
+        [T(A,0), None, None]
         >>> g[(2,5)]
-        [C(A,0), None, C(E,2)]
+        [T(A,0), None, T(E,2)]
         >>> g[(5,5)]
-        [C(B,0), C(C,1), C(E,2)]
+        [T(B,0), T(T,1), T(E,2)]
         >>> g[(6,5)]
-        [C(B,0), C(C,1), None]
+        [T(B,0), T(T,1), None]
         >>> # This channel fits in the hole left by the last one.
-        >>> g.add_channel(Channel((0, 5), (2, 5), None, "F"))
-        C(F,1)
+        >>> g.create_track(Track((0, 5), (2, 5), None, "F"))
+        T(F,1)
         >>> g[(0,5)]
-        [C(A,0), C(F,1), None]
+        [T(A,0), T(F,1), None]
         >>> g[(1,5)]
-        [C(A,0), C(F,1), None]
+        [T(A,0), T(F,1), None]
         >>> g[(2,5)]
-        [C(A,0), C(F,1), C(E,2)]
+        [T(A,0), T(F,1), T(E,2)]
         >>> g[(3,5)]
-        [C(A,0), None, C(E,2)]
+        [T(A,0), None, T(E,2)]
         >>> # Add another channel which causes a hole
-        >>> g.add_channel(Channel((0, 5), (6, 5), None, "G"))
-        C(G,3)
+        >>> g.create_track(Track((0, 5), (6, 5), None, "G"))
+        T(G,3)
         >>> g[(0,5)]
-        [C(A,0), C(F,1), None, C(G,3)]
+        [T(A,0), T(F,1), None, T(G,3)]
         >>> g[(1,5)]
-        [C(A,0), C(F,1), None, C(G,3)]
+        [T(A,0), T(F,1), None, T(G,3)]
         >>> g[(2,5)]
-        [C(A,0), C(F,1), C(E,2), C(G,3)]
+        [T(A,0), T(F,1), T(E,2), T(G,3)]
         >>> g[(3,5)]
-        [C(A,0), None, C(E,2), C(G,3)]
+        [T(A,0), None, T(E,2), T(G,3)]
         >>> g[(4,5)]
-        [C(B,0), C(C,1), C(E,2), C(G,3)]
+        [T(B,0), T(T,1), T(E,2), T(G,3)]
         >>> g[(5,5)]
-        [C(B,0), C(C,1), C(E,2), C(G,3)]
+        [T(B,0), T(T,1), T(E,2), T(G,3)]
         >>> g[(6,5)]
-        [C(B,0), C(C,1), None, C(G,3)]
+        [T(B,0), T(T,1), None, T(G,3)]
         >>> g[(7,5)]
         [None, None, None, None]
         """
-        assert ch.idx == None
+        assert t.idx == None
+        force_idx = idx
```

Comment:
I would probably have a `set_track` and a `add_track` -- only the set_track allows an idx value to be set?